perception/autoware_tensorrt_bevdet/src/bevdet_node.cpp

```diff
@@ -0,0 +1,261 @@
+    rclcpp::SensorDataQoS{}.keep_last(1).get_rmw_qos_profile());
+  sub_br_img_.subscribe(
+    this, "~/input/topic_img_back_right",
+    rclcpp::SensorDataQoS{}.keep_last(1).get_rmw_qos_profile());
+
+  sync_ = std::make_shared<Sync>(
+    MultiCameraApproxSync(10), sub_fl_img_, sub_f_img_, sub_fr_img_, sub_bl_img_, sub_b_img_,
+    sub_br_img_);
+
+  sync_->registerCallback(std::bind(&TRTBEVDetNode::callback, this, _1, _2, _3, _4, _5, _6));
+}
+
+void TRTBEVDetNode::startCameraInfoSubscription()
+{
+  // cams: ["CAM_FRONT_LEFT", "CAM_FRONT", "CAM_FRONT_RIGHT", "CAM_BACK_LEFT", "CAM_BACK",
+  // "CAM_BACK_RIGHT"]
+  sub_fl_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_front_left/camera_info", rclcpp::SensorDataQoS{}.keep_last(1),
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(0, msg); });
+
+  sub_f_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_front/camera_info", rclcpp::SensorDataQoS{}.keep_last(1),
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(1, msg); });
+
+  sub_fr_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_front_right/camera_info", rclcpp::SensorDataQoS{}.keep_last(1),
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(2, msg); });
+
+  sub_bl_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_back_left/camera_info", rclcpp::SensorDataQoS{}.keep_last(1),
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(3, msg); });
+
+  sub_b_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_back/camera_info", rclcpp::SensorDataQoS{}.keep_last(1),
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(4, msg); });
+
+  sub_br_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_back_right/camera_info", rclcpp::SensorDataQoS{}.keep_last(1),
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(5, msg); });
+}
+
+void TRTBEVDetNode::callback(
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_f_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fr_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_bl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_b_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_br_img)
+{
+  std::vector<cv::Mat> imgs;
+  auto clone_and_resize = [&](const sensor_msgs::msg::Image::ConstSharedPtr & msg) -> cv::Mat {
+    cv::Mat img = cv_bridge::toCvShare(msg, "bgr8")->image.clone();
+    if (img.size() != cv::Size(img_w_, img_h_)) {
+      cv::resize(img, img, cv::Size(img_w_, img_h_));  // img_w_ = 1600, img_h_ = 900
+    }
+    return img;
+  };
```

Comment:
If there is no specific reason to do this, then we can make this expression into a method.